guideline/static-context-provider.md

```diff
@@ -0,0 +1,87 @@
+# Creating an OFREP client provider
+
+OpenFeature Remote Evaluation Protocol (OFREP) is an API specification for feature flagging that allows the use of generic providers to connect to any feature flag management systems that supports the protocol.
+
+In this document, we will specify how to write an OFREP provider using the [static-context-paradigm](https://openfeature.dev/specification/glossary/#static-context-paradigm) that is used on client side applications typically operate in the context of a single user. 
+We will keep the document language agnostic.
+
+**Pre-requisite:**
+- Understanding of [general provider concepts](https://openfeature.dev/docs/reference/concepts/provider/)
+- Understanding of the [OFREP](../../README.md)
+- Understanding of the [OFREP OpenAPI specification](../../service/openapi.yaml)
+
+## Constructor
+An implementation of an OFREP client provider should allow in the creation of the provider to take as options:
+- `baseURL`: The base URL of the [flag management system](https://openfeature.dev/specification/glossary#flag-management-system).  
+  This should be the base of the URL pointing before the `/ofrep` namespace of the API.
+  - In the constructor, the provider should check if the `baseURL` is a valid URL and return an error if the URL is invalid.
+- `headers`: The headers to use when calling the OFREP endpoints *(ex:`Authorization`, Custom headers, etc ...)*.
+- `pollInterval`: The polling interval defining when to call again the flag management system.
+  - If `pollInterval` is equals to 0, polling will be disabled.
+
+## Initialize the provider
+An implementation of an OFREP client provider should start with an initialization of the provider.
+
+The `initialize()` function should follow those steps:
+1. Make a POST request to the `/ofrep/v1/evaluate/flags` endpoint with the evaluation context in the body.
+
+   **Request body example**:
+   ```json
+    {
+      "context": {
+        "targetingKey": "f021d0f9-33b7-4b22-b0bd-9fec66ba1d7d",
+        "firstname": "foo",
+        "lastname": "bar",
+        "email": "foo.bar@ofrep.dev"
+      }
+    }
+   ```
+
+    - If the endpoint returns an error, the `initialize()` function must error and exit.  
+    - If the request is successful:
+      - We should store in a local cache all of the flags evaluation details returned by the API. 
+      - We should store the `ETag` header in the provider to be able to send it back later.
+2. If `pollInterval` is more than `0`, we should start a polling mechanism to periodically check for flag changes *([See polling section](#polling))*.
```

Comment:
Should we define if I send an invalid `pollInterval`?

Reply:
I am not sure what you mean by an invalid `pollInterval`?

Reply:
When someone sends a `pollInterval` of negative or zero.